hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/metadata/FlinkHoodieBackedTableMetadataWriter.java

```diff
@@ -199,6 +199,18 @@ protected HoodieData<HoodieRecord> getExpressionIndexRecords(List<Pair<String, P
     throw new HoodieNotSupportedException("Flink metadata table does not support expression index yet.");
   }
 
+  protected void bulkInsertAndCommit(BaseHoodieWriteClient<?, List<HoodieRecord>, ?, List<WriteStatus>> writeClient, String instantTime, List<HoodieRecord> preppedRecordInputs,
+                                     Option<BulkInsertPartitioner> bulkInsertPartitioner) {
+    List<WriteStatus> writeStatusJavaRDD = writeClient.bulkInsertPreppedRecords(preppedRecordInputs, instantTime, bulkInsertPartitioner);
+    writeClient.commit(instantTime, writeStatusJavaRDD);
```

Comment:
Maybe we bydefault always make `bulkInsertPreppedRecords` and `upsertPreppedRecords` complete the commits because they are only used for MDT so we can remove these override methods.

Reply:
We might use prepped APIs in spark-sql writer flows for data table as well. 
So, don't wanna special case bcoz of that.

Reply:
then add inline completion API for it, let's eliminate these override methods in metadata writer.

Reply:
these are public apis right. why add additional overloaded method that can auto commit. 
lets keep it clean. 
only option for the user of WriteClient is to call writeClient.commit() 
except few cases (bootstrap, partition TTL). 

I just don't want to keep it open, then devs can start mis-using it. and it may not really worth the effort we took now to deprecate the auto commit